ServantHDBC/src/Api/Tenant.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+
+
+module Api.Tenant (TenantAPI, tenantHandlers) where
+
+import Control.Monad.Trans.Except
+import Servant
+
+type Tenant = String
+type TenantAPI = "tenants" :> 
+                         ("new" :> Post '[JSON] Tenant
+                          :<|> Capture "x" Int :> Get '[JSON] Tenant
```

Comment:
@saurabhnanda I guess the URL would be (/tenants/:x)   x -> is the tenant id?

I would personally would like to have these as separate lines...

```
type TenantAPI = "tenants" :> 
                         ("new" :> Post '[JSON] Tenant
                          :<|> Capture "x" Int :> Get '[JSON] Tenant
                          :<|> Capture "x" Int :> "activate" :> Post '[JSON] Tenant)
```

I like this...

```
type TenantAPI 
  = "tenants" :> "new" :> Post '[JSON] Tenant
  :<|> "tenants" :> Capture "x" Int :> Get '[JSON] Tenant
  :<|> "tenants" :> Capture "x" Int :> "activate" :> Post '[JSON] Tenant
```

if repeated "tenants" is the problem then...

```
type TenantAPI' 
  = "new" :> Post '[JSON] Tenant
  :<|> Capture "x" Int :> Get '[JSON] Tenant
  :<|> Capture "x" Int :> "activate" :> Post '[JSON] Tenant

type TenantAPI = "tenants" :> TenantAPI'
```

I like to avoid ()'s what do you think?